### PR TITLE
Added checks and better error reporting for common issues

### PR DESCRIPTION
**Pull request by jakrams**

### CODE REVIEW

File: ArrtSource/Utils/CheckRequirements.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    }
+} // namespace
+
+bool IsVideoDecoderAvailable(std::string& details)
+{
+    static bool bCheckedAlready = false;
+    if (bCheckedAlready)
+        return true;
+
+    HRESULT hr = ::MFStartup(MF_VERSION, MFSTARTUP_FULL);
```
Comment (FlorianBorn71) [inline ArrtSource/Utils/CheckRequirements.cpp:108]: no shutdown required?

Comment (jakrams) [reply ArrtSource/Utils/CheckRequirements.cpp:108]: See line 121

Comment (FlorianBorn71) [reply ArrtSource/Utils/CheckRequirements.cpp:108]: Oh missed it ;-)

**Review by FlorianBorn71 — APPROVED** [review]

**Pull request merged**